src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -175,14 +184,36 @@ const SignOrExecuteForm = ({
     return id
   }
 
+  const onRelay = async () => {
+    const { gasLimit } = getTxOptions(advancedParams, currentChain)
+
+    const [, createdTx] = assertDependencies()
+
+    let txData
+    try {
+      txData = await prepareRelayTxData(createdTx, safe, proposeTx)
+    } catch (err) {
+      console.error('Error signing relay tx data', err)
+      return
+    }
+
+    try {
+      const relayResponse = await sponsoredCall({ chainId: safe.chainId, to: safeAddress, data: txData.data, gasLimit })
+      await dispatchTxRelay(relayResponse.taskId, txData.txId)
```

Comment:
If the transaction is already signed, it will not get proposed again in `prepareRelayTxData` and thus `txData.txId` will be undefined. You should use `txData.txId || txId` instead

Reply:
I've changed this logic. If the tx is already signed we use that `txId`